libcodechecker/libhandlers/store.py

```diff
@@ -310,124 +297,31 @@ def main(args):
-                        check_commands.append(metadata_dict['command'])
-                    if 'timestamps' in metadata_dict:
-                        check_durations.append(
-                            float(metadata_dict['timestamps']['end'] -
-                                  metadata_dict['timestamps']['begin']))
-                    if 'skip_data' in metadata_dict:
-                        # Save previously stored skip data for sending to the
-                        # database, to ensure skipped headers are actually
-                        # skipped --- 'analyze' can't do this.
-                        handle, path = tempfile.mkstemp()
-                        with os.fdopen(handle, 'w') as tmpf:
-                            tmpf.write('\n'.join(metadata_dict['skip_data']))
-
-                        skip_handlers.append(
-                            skiplist_handler.SkipListHandler(path))
-                        os.remove(path)
-            else:
-                metadata_dict = empty_metadata
-
-            _, _, files = next(os.walk(input_path), ([], [], []))
-            for f in files:
-                if not f.endswith(".plist"):
-                    continue
-
-                items.append((os.path.join(input_path, f),
-                              context, metadata_dict, compile_commands,
-                              server_data))
-
-    if len(check_commands) == 0:
-        command = ' '.join(sys.argv)
-    elif len(check_commands) == 1:
-        command = ' '.join(check_commands[0])
-    else:
-        command = "multiple analyze calls: " +\
-                  '; '.join([' '.join(com) for com in check_commands])
+    workspace = util.get_default_workspace()
+    try:
+        os.makedirs(workspace)
+    except os.error:
+        # Workdir already exists.
+        pass
 
     # setup connection to the remote server
-    client = setup_client(args.host,
-                          args.port, '/')
+    client = setup_client(args.host, args.port, '/')
 
     LOG.debug("Initializing client connecting to " +
               str(args.host) + ":" + str(args.port) + " done.")
-    context.run_id = client.addCheckerRun(command,
-                                          args.name,
-                                          context.version,
-                                          args.force)
 
-    # Send previously collected skip information to the server.
-    LOG.debug("Storing skip information")
-    for skip_handler in skip_handlers:
-        if not client.addSkipPath(context.run_id, skip_handler.get_skiplist()):
-            LOG.debug("Adding skip path failed!")
-
-    # TODO: This is a hotfix for a data race problem in storage.
-    # Currently removal of build actions is based on plist files which lack
-    # build command that is needed for unambiguous deletion.
-    args.jobs = 1
-
-    pool = multiprocessing.Pool(args.jobs)
+    _, zip_file = tempfile.mkstemp('.zip', dir=workspace)
```

Comment:
If this file is only temporary and gets cleaned up all the time, wouldn't we be better off putting it under `/tmp`? The concept of _workspace_ has been marked for removal in version 6 in favour of `config-dir` and `output-dir` and such.

Perhaps even one could use the analyze-output folder given to `store` instead of resorting to this environmental "constant".